From cb5f8f1c0b24f1b215b2bb5eb6f9a8e16d728ce2 Mon Sep 17 00:00:00 2001
From: Michael Neeley <micneeley14@gmail.com>
Date: Mon, 14 Jun 2021 07:55:07 -0400
Subject: [PATCH] fix: turns off automerging new people (#1989)

* fix: turns off automerging new people

* fix

* tests and snaps
---
 .../src/auth/__tests__/index.tests.js                 |  4 +++-
 .../__tests__/__snapshots__/data-source.tests.js.snap |  8 +-------
 .../src/people/data-source.js                         | 11 ++++++++---
 3 files changed, 12 insertions(+), 11 deletions(-)

diff --git a/packages/apollos-data-connector-rock/src/auth/__tests__/index.tests.js b/packages/apollos-data-connector-rock/src/auth/__tests__/index.tests.js
index edf11916f8..2b5fbf7b8b 100644
--- a/packages/apollos-data-connector-rock/src/auth/__tests__/index.tests.js
+++ b/packages/apollos-data-connector-rock/src/auth/__tests__/index.tests.js
@@ -224,11 +224,12 @@ describe('Auth', () => {
     });
 
     it('creates user profile', async () => {
+      context.dataSources.Person.create = jest.fn(() => 35);
       const result = await context.dataSources.Auth.createUserProfile({
         email: 'isaac.hardy@newspring.cc',
       });
 
-      expect(result).toEqual('35');
+      expect(result).toEqual(35);
     });
 
     it('throws error in createUserProfile', async () => {
@@ -276,6 +277,7 @@ describe('Auth', () => {
           }
         }
       `;
+      context.dataSources.Person.create = jest.fn(() => 1);
 
       const rootValue = {};
 
diff --git a/packages/apollos-data-connector-rock/src/people/__tests__/__snapshots__/data-source.tests.js.snap b/packages/apollos-data-connector-rock/src/people/__tests__/__snapshots__/data-source.tests.js.snap
index 26944d28b7..fd5845b834 100644
--- a/packages/apollos-data-connector-rock/src/people/__tests__/__snapshots__/data-source.tests.js.snap
+++ b/packages/apollos-data-connector-rock/src/people/__tests__/__snapshots__/data-source.tests.js.snap
@@ -7,19 +7,13 @@ Array [
   Array [
     "/People",
     Object {
-      "BirthDay": 1,
-      "BirthMonth": 2,
-      "BirthYear": 2020,
-      "FirstName": "Vincent",
-      "Gender": 1,
+      "Gender": 0,
       "IsSystem": false,
     },
   ],
 ]
 `;
 
-exports[`Person creates a profile 2`] = `Object {}`;
-
 exports[`Person gets person from aliasId: The call to fetch the alias id 1`] = `
 Array [
   Array [
diff --git a/packages/apollos-data-connector-rock/src/people/data-source.js b/packages/apollos-data-connector-rock/src/people/data-source.js
index 30c588d881..e8555fd577 100644
--- a/packages/apollos-data-connector-rock/src/people/data-source.js
+++ b/packages/apollos-data-connector-rock/src/people/data-source.js
@@ -65,13 +65,18 @@ export default class Person extends RockApolloDataSource {
     return null;
   };
 
-  create = (profile) => {
+  create = async (profile) => {
     const rockUpdateFields = this.mapApollosFieldsToRock(profile);
-    return this.post('/People', {
+    // auto-merge functionality is compromised
+    // we are creating a new user and patching them with profile details
+    const id = await this.post('/People', {
       Gender: 0, // required by Rock. Listed first so it can be overridden.
-      ...rockUpdateFields,
       IsSystem: false, // required by rock
     });
+    await this.patch(`/People/${id}`, {
+      ...rockUpdateFields,
+    });
+    return id;
   };
 
   mapGender = ({ gender }) => {
